tests/shared/UserLogonRequestTestDefs.js

```diff
@@ -0,0 +1,517 @@
+    jqUnit.assertDeepEq("The current active GPII key is as expected - " + expected, expected, lifecycleManager.getActiveSessionGpiiKeys());
+};
+
+gpii.tests.userLogonRequest.testLoginResponse = function (data, gpiiKey) {
+    jqUnit.assertEquals("Response is correct", "User with GPII key " +
+        gpiiKey + " was successfully logged in.", data);
+};
+
+gpii.tests.userLogonRequest.testLogoutResponse = function (data, gpiiKey) {
+    jqUnit.assertEquals("Response is correct", "User with GPII key " +
+        gpiiKey + " was successfully logged out.", data);
+};
+
+gpii.tests.userLogonRequest.modelChangeChecker = function (result, expectedType, expectedInProgress, expectedGpiiKey) {
+    jqUnit.assertEquals("Checking type of model change", expectedType, result.type);
+    jqUnit.assertEquals("Checking inProgress of model change", expectedInProgress, result.inProgress);
+    jqUnit.assertEquals("Checking gpiiKey of model change", expectedGpiiKey, result.gpiiKey);
+};
+
+gpii.tests.userLogonRequest.buildTestDefs = function (testDefs) {
+    return fluid.transform(testDefs, function (testDef) {
+        return fluid.extend(true, {
+            config: {
+                // The custom config file is to config the debounce time at proximityTriggered endpoint to 3 seconds
+                // rather than using the default 1.5 seconds. This is to work around an issue with testing a following
+                // request that occurs < the debounce time. In this test, the following request was sent after waiting
+                // for 1 second. However in the reality with all other running processes, CPU processing power etc,
+                // this request is usually sent a bit more than after 1 second, occassionally even more than 1.5 seconds,
+                // which causes the test to fail. Setting the debounce time to 3 seconds provides more buffering time
+                // for the following request to send.
+                configName: "gpii.tests.acceptance.userLogon.config",
+                configPath: "%gpii-universal/tests/configs"
+            },
+            gradeNames: ["gpii.tests.userLogonRequest.testCaseHolder", "gpii.test.integration.testCaseHolder.linux"],
+            gpiiKey: testDefs.gpiiKey || gpii.tests.userLogonRequest.gpiiKey
+        }, testDef);
+    });
+};
+
+
+gpii.tests.userLogonRequest.testDefs = [{
+    name: "Testing standard proximityTriggered login and logout",
+    expect: 29,
+    sequence: [{
+        // The initial active GPII key is "noUser"
+        func: "gpii.tests.userLogonRequest.verifyActiveGpiiKey",
+        args: ["{lifecycleManager}", ["noUser"]]
+    }, {
+        // 1. 1nd proximityTriggered request to key in testUser1
+        func: "gpii.tests.invokePromiseProducer",
+        args: ["{lifecycleManager}.performProximityTriggered", [gpii.tests.userLogonRequest.gpiiKey], "{that}"]
+    }, {
+        changeEvent: "{lifecycleManager}.applier.modelChanged",
```

Comment:
Sheer luck that this sequence works in the presence of https://issues.fluidproject.org/browse/FLUID-5502 ! It will be safer to instead register a single listener which assembles all successive changes into an array and then check the array just once with jqUnit.assertDeepEq

Reply:
Please add a note in this fixture file that there is an implementation risk due to FLUID-5502 but we believe that this is currently reliable because each of the model changes is triggered asynchronously